pkgs/development/tools/misc/blackfire/default.nix

```diff
@@ -59,7 +46,58 @@ stdenv.mkDerivation rec {
     description = "Blackfire Profiler agent and client";
     homepage = "https://blackfire.io/";
     license = licenses.unfree;
-    maintainers = with maintainers; [ jtojnar ];
-    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ jtojnar shyim ];
+    platforms = [ "x86_64-linux" "aarch64-linux" "i686-linux" "x86_64-darwin" "aarch64-darwin" ];
   };
-}
+
+  src = sources.${stdenv.hostPlatform.system};
+
+in if stdenv.isLinux then
+  stdenv.mkDerivation rec {
+    inherit pname version passthru meta src;
+
+    nativeBuildInputs = [ dpkg autoPatchelfHook ];
+
+    dontUnpack = true;
+
+    installPhase = ''
+      runHook preInstall
+      dpkg-deb -x $src $out
+      mv $out/usr/* $out
+      rmdir $out/usr
+      # Fix ExecStart path and replace deprecated directory creation method,
+      # use dynamic user.
+      substituteInPlace "$out/lib/systemd/system/blackfire-agent.service" \
+        --replace '/usr/' "$out/" \
+        --replace 'ExecStartPre=/bin/mkdir -p /var/run/blackfire' 'RuntimeDirectory=blackfire' \
+        --replace 'ExecStartPre=/bin/chown blackfire: /var/run/blackfire' "" \
+        --replace 'User=blackfire' 'DynamicUser=yes' \
+        --replace 'PermissionsStartOnly=true' ""
+      # Modernize socket path.
+      substituteInPlace "$out/etc/blackfire/agent" \
+        --replace '/var/run' '/run'
+      runHook postInstall
+    '';
+  }
+else
+
+  stdenv.mkDerivation rec {
+    inherit pname version passthru meta src;
```

Comment:
Most of the derivation is the same so I wonder if it would not be nicer to just use a single `mkDerivation` and move the conditional inside `nativeBuildInputs` and `installPhase`. I do not have a strong preference either way, though.